src/dsp_tools/utils/excel_to_json/properties.py

```diff
@@ -0,0 +1,457 @@
+import importlib.resources
+import json
+import warnings
+from typing import Any, Optional
+import jsonpath_ng.ext
+import jsonschema
+import regex
+import pandas as pd
+import numpy as np
+
+import dsp_tools.utils.excel_to_json.utils as utl
+from dsp_tools.models.exceptions import UserError
+
+languages = ["en", "de", "fr", "it", "rm"]
+language_label_col = ["label_en", "label_de", "label_fr", "label_it", "label_rm"]
+mandatory_properties = ["name", "object", "gui_element"]
+
+
+def _search_validation_error(
+    list_properties: list[dict[str, Any]], excel_filename: str, validation_error: jsonschema.ValidationError
+) -> str:
+    """
+    This function takes a list of properties, which are formatted as a dictionary and an error which was raised by the
```

Comment:
Here, I think that the docstring doesn't give a good description of what actually happens. 

- The properties are not "formatted" as dictionary. Rather, they have been transformed from Excel-structure into JSON-structure. So "structured" is perhaps better than "formatted".
- "error which was raised by the calling function" -->" error from the JSON schema validation" would be more telling
- "which exact location of the property caused the error". Perhaps better: "which location in the Excel file caused the error"